drivers/I2C.cpp

```diff
@@ -41,16 +42,23 @@ I2C::I2C(PinName sda, PinName scl) :
     _sda = sda;
     _scl = scl;
     recover(sda, scl);
-    i2c_init(&_i2c, _sda, _scl);
+    i2c_init(&_i2c, sda, scl, false);
+    frequency(_hz);
+
     // Used to avoid unnecessary frequency updates
     _owner = this;
     unlock();
 }
 
+I2C::~I2C()
+{
+    i2c_free(&_i2c);
+}
+
 void I2C::frequency(int hz)
 {
     lock();
-    _hz = hz;
+    _hz = (uint32_t)hz;
```

Comment:
*nitpick:* Should this be guarded to prevent setting of negative values ?

Reply:
Fixed
`MBED_ASSERT(_hz > 0);` added for both master and slave driver